lms/services/application_instance_getter.py

```diff
@@ -70,6 +71,18 @@ def provisioning_enabled(self):
             provisioning = False
         return provisioning
 
+    def canvas_sections_enabled(self):
+        """Return True if the application instance has Canvas sections is enabled."""
+        if not self._section_groups_feature_flag:
+            return False
+
+        try:
+            app_instance = self._get_by_consumer_key()
+        except ConsumerKeyError:
+            return False
+
+        return bool(app_instance.developer_key and app_instance.canvas_sections_enabled)
```

Comment:
This is the "main" logic for "Is Canvas sections enabled?" The application instance has to both have a developer key and have `canvas_sections_enabled`.

It _is_ possible to have a developer key but _not_ have `canvas_sections_enabled`: this will happen for preexisting app instances that have a developer key.

Also, sections is off if the feature flag is off. The fact that `ApplicationInstanceGetter` is a service turned out to be quite handy 'cos the service factory can inject the feature flag.